### PR TITLE
go/common/keyformat: Ensure consensus and runtime prefixes are unique

### PR DESCRIPTION
**Pull request by peternose**

### CODE REVIEW

**Review by kostko — COMMENTED** [review]
Could you maybe change this so that you can basically specify a namespace where it needs to be unique? Consensus looks fine, but runtime as it is currently doesn't make sense (e.g. history db keys and runtime transaction tree don't need to have unique prefixes -- those only need to be unique within that IO tree).

**Comment by peternose** [comment]
> history db keys and runtime transaction tree don't need to have unique prefixes

Wasn't sure about that and wanted to simplify. Will add namespaces.

File: go/storage/mkvs/db/badger/badger.go [hunk 1]
```diff
@@ -32,37 +32,37 @@ var (
 	// nodeKeyFmt is the key format for nodes (node hash).
 	//
 	// Value is serialized node.
-	nodeKeyFmt = keyformat.New(0x00, &hash.Hash{})
+	nodeKeyFmt = keyformat.Consensus(0x00, &hash.Hash{})
```
Comment (kostko) [inline go/storage/mkvs/db/badger/badger.go:35]: This should not be consensus, this is another database for storing runtime and consensus state (both use the same but these keys here are internal). I don't think we should hardcode "runtime history" and all the database backend namespaces in the key format package though?

Could we have a notion of namespaces that are specified externally? E.g. something like:
```golang
// Somewhere we would define the namespace.
var Consensus = keyformat.NewNamespace()
// Then we can uset the namespace to create new key formats.
var fooFmt = Consensus.New(0x00, ...)
```
This (or similar) would make it more flexible to use in external packages without the need to change the `keyformat` package.

Comment (peternose) [reply go/storage/mkvs/db/badger/badger.go]: You are right. I reverted some changes and added namespace constructor (with a namespace name to have clearer panic message). Now there are 4 namespaces: consensus, runtime db, runtime tx, and badger. Key formats defined in tests and in migration don't belong to any namespace (duplicates are allowed).

File: go/storage/mkvs/db/badger/badger.go [hunk 2]
```diff
@@ -29,40 +29,43 @@ const (
 )
 
 var (
+	// keyFormat is the namespace for the runtime transaction key formats.
```
Comment (kostko) [inline go/storage/mkvs/db/badger/badger.go:32]:
```suggestion
	// keyFormat is the namespace for the badger database key formats.
```

File: go/consensus/api/api.go [hunk 3]
```diff
@@ -65,6 +66,9 @@ var (
 	SystemMethods = map[transaction.MethodName]struct{}{
 		MethodMeta: {},
 	}
+
+	// KeyFormat is the namespace for the consensus key formats.
```
Comment (kostko) [inline go/consensus/api/api.go:70]:
```suggestion
	// KeyFormat is the namespace for the consensus state key formats.
```

**Review by kostko — COMMENTED** [review]
Looks good now, just two minor comment fixes.

**Review by kostko — APPROVED** [review]

**Pull request merged**